packages/cli/generation/local-generation/local-workspace-runner/src/dynamic-snippets/csharp/DynamicSnippetsCsharpTestGenerator.ts

```diff
@@ -0,0 +1,115 @@
+        <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.13.0"/>
+        <PackageReference Include="NUnit" Version="4.3.2"/>
+        <PackageReference Include="NUnit.Analyzers" Version="4.6.0">
+            <PrivateAssets>all</PrivateAssets>
+            <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+        </PackageReference>
+        <PackageReference Include="coverlet.collector" Version="6.0.4">
+            <PrivateAssets>all</PrivateAssets>
+            <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+        </PackageReference>
+    </ItemGroup>
+
+    <ItemGroup>
+        <ProjectReference Include="..\\*\\*.csproj" Exclude="..\\*\\*.DynamicSnippets.csproj;..\\*\\*.Test.csproj" />
+    </ItemGroup>
+</Project>`;
+
+export class DynamicSnippetsCsharpTestGenerator {
+    private dynamicSnippetsGenerator: DynamicSnippetsGenerator;
+
+    constructor(
+        private readonly context: TaskContext,
+        private readonly ir: dynamic.DynamicIntermediateRepresentation,
+        private readonly generatorConfig: FernGeneratorExec.GeneratorConfig
+    ) {
+        this.dynamicSnippetsGenerator = new DynamicSnippetsGenerator({
+            ir: convertIr(this.ir),
+            config: this.generatorConfig
+        });
+    }
+
+    public async generateTests({
+        outputDir,
+        requests
+    }: {
+        outputDir: AbsoluteFilePath;
+        requests: dynamic.EndpointSnippetRequest[];
+    }): Promise<void> {
+        this.context.logger.debug("Generating dynamic snippet tests...");
+        const absolutePathToOutputDir = await this.initializeProject(outputDir);
+        for (const [idx, request] of requests.entries()) {
+            try {
+                const response = await this.dynamicSnippetsGenerator.generate(
+                    convertDynamicEndpointSnippetRequest(request),
+                    {
+                        config: {
+                            fullStyleClassName: `Example${idx}`
+                        } as Config,
+                        style: Style.Full
+                    }
+                );
+                const dynamicSnippetFilePath = this.getTestFilePath({ absolutePathToOutputDir, idx });
+                await mkdir(path.dirname(dynamicSnippetFilePath), { recursive: true });
+                await writeFile(dynamicSnippetFilePath, response.snippet);
+            } catch (error) {
+                this.context.logger.error(
+                    `Failed to generate dynamic snippet for endpoint ${JSON.stringify(request.endpoint)}: ${error}`
+                );
+            }
+        }
+        this.context.logger.debug("Done generating dynamic snippet tests");
+    }
+
+    private async initializeProject(outputDir: AbsoluteFilePath): Promise<AbsoluteFilePath> {
+        const absolutePathToOutputDir = join(outputDir, RelativeFilePath.of("src/SeedApi.DynamicSnippets"));
```

Comment:
`src/SeedApi.DynamicSnippets` this name is still hardcoded.

Reply:
That's actually OK in this context - we're just using this to verify that the project compiles so the project name doesn't actually need to match. This keeps the implementation a little simpler.